lib/ansible/modules/source_control/gitlab_tag.py

```diff
@@ -0,0 +1,268 @@
+requirements:
+    - python-gitlab python module
+options:
+    server_url:
+        description:
+            - Url of Gitlab server, with protocol (http or https).
+        required: true
+    validate_certs:
+        description:
+            - When using https if SSL certificate needs to be verified.
+        required: false
+        default: true
+        aliases:
+            - verify_ssl
+    login_user:
+        description:
+            - Gitlab user name.
+        required: false
+        default: null
+    login_password:
+        description:
+            - Gitlab password for login_user
+        required: false
+        default: null
+    login_token:
+        description:
+            - Gitlab token for logging in.
+        required: false
+        default: null
+    group:
+        description:
+            - The name of the group of which the projects belongs to.
+            - If the project belongs to a user, the username should be used for this.
+        required: true
+        default: null
+    project:
+        description:
+            - The name of the project on which the hook needs to be added or updated.
+        required: true
+        default: null
+    name:
+        description:
+            - The name of the tag you want to create
+        required: true
+        default: null
+    ref:
+        description:
+            - Create tag using commit SHA, another tag name, or branch name
+        required: false
+        default: "master"
+    description:
+        description:
+            - A description of the tag.
+        required: false
+        default: null
+    state:
+        description:
+            - create or delete hooks.
+        required: false
+        default: "present"
+        choices: ["present", "absent"]
+
+'''
+
+EXAMPLES = '''
+- name: "Create tags"
+  local_action:
+    module: gitlab_tag
+    server_url="http://gitlab.dj-wasabi.local"
+    validate_certs=false
+    login_token="WnUzDsxjy8230-Dy_k"
+    group: my_first_group
+    project: my_first_project
+    name: v1.0
+    ref: master
+    description: My First Release
+    state: present
+
+- name: "Delete tag"
+  local_action:
+    module: gitlab_tag
+    server_url="http://gitlab.dj-wasabi.local"
+    validate_certs=false
+    login_token="WnUzDsxjy8230-Dy_k"
+    group: my_first_group
+    project: my_first_project
+    name: v0.9a
+    ref: master
+    description: My very bad release
+    state: absent
+
+'''
+
+RETURN = '''# '''
+
+try:
+    import gitlab
+    HAS_GITLAB_PACKAGE = True
+except:
+    HAS_GITLAB_PACKAGE = False
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.pycompat24 import get_exception
+
+
+class GitLabTag(object):
+    def __init__(self, module, git):
+        self._module = module
+        self._gitlab = git
+        self.projectObject = None
+        self.tagObject = None
+
+    def existsTag(self, name):
+        """Validate is tag exists"""
+        project = self.projectObject
```

Comment:
If something calls `existsTag` before `getProject` is called, `project` will hold `None`, and will then cause problems. I think there should be a `GitLabProject` class (which should also go into the `ansible.module_utils` namespace) which should be passed in to this class through the constructor.